### PR TITLE
Flaky test: TestAgentAttest

### LINKED ISSUE #3716: Flaky test: TestAttestAgent
```
--- FAIL: TestAttestAgent (0.64s)
    --- FAIL: TestAttestAgent/attest_with_join_token_only_works_once (0.03s)
        logs.go:18: 
            	Error Trace:	/home/runner/work/spire/spire/pkg/server/api/agent/v1/logs.go:18
            	            				/home/runner/work/spire/spire/pkg/server/api/agent/v1/service_test.go:3011
            	            				/home/runner/work/spire/spire/pkg/server/api/agent/v1/service_test.go:3058
            	Error:      	Not equal: 
            	            	expected: []spiretest.LogEntry{spiretest.LogEntry{Level:0x2, Message:"Invalid argument: failed to attest: join token does not exist or has already been used", Data:logrus.Fields{"node_attestor_type":"join_token"}}, spiretest.LogEntry{Level:0x4, Message:"API accessed", Data:logrus.Fields{"node_attestor_type":"join_token", "status":"error", "status_code":"InvalidArgument", "status_message":"failed to attest: join token does not exist or has already been used", "type":"audit"}}}
            	            	actual  : []spiretest.LogEntry{spiretest.LogEntry{Level:0x2, Message:"Invalid argument: failed to attest: join token does not exist or has already been used", Data:logrus.Fields{"node_attestor_type":"join_token"}}, spiretest.LogEntry{Level:0x4, Message:"API accessed", Data:logrus.Fields{"node_attestor_type":"join_token", "status":"error", "status_code":"InvalidArgument", "status_message":"failed to attest: join token does not exist or has already been used", "type":"audit"}}, spiretest.LogEntry{Level:0x4, Message:"API accessed", Data:logrus.Fields{"agent_id":"spiffe://example.org/spire/agent/join_token/test_token", "node_attestor_type":"join_token", "status":"success", "type":"audit"}}}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,2 +1,2 @@
            	            	-([]spiretest.LogEntry) (len=2) {
            	            	+([]spiretest.LogEntry) (len=3) {
            	            	  (spiretest.LogEntry) {
            	            	@@ -18,2 +18,12 @@
            	            	   }
            	            	+ },
            	            	+ (spiretest.LogEntry) {
            	            	+  Level: (logrus.Level) 4,
            	            	+  Message: (string) (len=12) "API accessed",
            	            	+  Data: (logrus.Fields) (len=4) {
            	            	+   (string) (len=8) "agent_id": (string) (len=54) "spiffe://example.org/spire/agent/join_token/test_token",
            	            	+   (string) (len=18) "node_attestor_type": (string) (len=10) "join_token",
            	            	+   (string) (len=6) "status": (string) (len=7) "success",
            	            	+   (string) (len=4) "type": (string) (len=5) "audit"
            	            	+  }
            	            	  }
            	Test:       	TestAttestAgent/attest_with_join_token_only_works_once
            	Messages:   	unexpected logs
FAIL
```

### PR DESCRIPTION
**Pull request by MarcosDY**
Solve flaky tests caused by edge cases where audit logs from previous calls was not successfully cleaned

**Which issue this PR fixes**
fixes #3716

### CODE REVIEW

**Review by amartinezfayo — APPROVED** [review]
Thank you Marcos, LGTM!
